src/batch-client/index.ts

```diff
@@ -433,6 +436,57 @@ export class ZimbraBatchClient {
 			return ids ? ids.split(',') : [];
 		});
 
+	// For offline Drafts
+	public attach = (files: any, { ...message }) => {
+		const promises = castArray(files).map(file => this.localStoreClient.attach({ file }));
+
+		return Promise.all(promises).then(attached => {
+			// After all uploads have completed, add attachments to the message and save it as a draft.
+			attached.forEach((attachment, index) => {
+				if (attachment) {
+					const file = files[index];
```

Comment:
i think we should move const file = files[index]; inside inline if

Reply:
Code refactored and moved to zm-x-web